### PR TITLE
Set up GitHub Actions

### LINKED ISSUE #173: Appveyor/Nuget Package Release
@merbla Is the appveyor build still set up to work on the master branch? Is there anything further that needs to be done to push this to nuget other than merging changes into master? Thank you for your time.

### PR DESCRIPTION
**Pull request by nblumhardt**
This setup is via https://github.com/serilog-tracing/serilog-tracing.

I've gone for the quickest thing I could pull together, happy to tweak it here, or just merge/debug and leave it in your hands, @EEParker and @VictorioBerra :-)

Closes #173

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.0.x
+      - name: Compute build number
+        run: |
+          echo "CI_BUILD_NUMBER=$(($CI_BUILD_NUMBER_BASE+200))" >> $GITHUB_ENV
+      - name: Build and Publish
+        env:
+          DOTNET_CLI_TELEMETRY_OPTOUT: true
+          NUGET_API_KEY: ${{ secrets.NUGET_API_KEY }}
```
Comment (EEParker) [inline .github/workflows/ci.yml:32]: Who has access to this repositories secrets? It looks like that will be required to make this work.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.0.x
+      - name: Compute build number
+        run: |
+          echo "CI_BUILD_NUMBER=$(($CI_BUILD_NUMBER_BASE+200))" >> $GITHUB_ENV
```
Comment (VictorioBerra) [inline .github/workflows/ci.yml:28]: Just curious, why +200?

Comment (EEParker) [reply .github/workflows/ci.yml:28]: looks like a copy/paste of https://github.com/serilog-tracing/serilog-tracing/blob/dev/.github/workflows/ci.yml

Comment (EEParker) [reply .github/workflows/ci.yml:28]: It looks like it is just to keep this incrementing and not start at 0. 
image

**Comment by EEParker** [comment]
This looks like a good start, I would also like to add something to generate release notes for tags (release versions)

For example:

### ci-release.yml
```yml
on:
  push:
    tags:
    - '*'

# [ include content from dev build ci] [...]
    steps:
   # [...]
   - name: Create Release
      uses: ncipollo/release-action@v1
      with: 
        artifacts: "*.nupkg"
        generateReleaseNotes: true
```

File: .gitignore [hunk 3]
```diff
@@ -239,4 +239,6 @@ _Pvt_Extensions
 
 sample/Sample/out/
 
-.DS_Store
\ No newline at end of file
+.DS_Store
+.gitignore
```
Comment (EEParker) [inline .gitignore:243]: can `.idea/` be added to `.gitignore` instead of ignoring `.gitignore`

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nblumhardt) [reply .github/workflows/ci.yml:29]: I've set this one up already 👍

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nblumhardt) [reply .github/workflows/ci.yml]: Fixed, thanks

File: .gitignore (hunk 3 above)
Comment (nblumhardt) [reply .gitignore]: Argh :-) - fixed 👍

**Comment by nblumhardt** [comment]
Hopefully what's there is enough to get the ball rolling now; generating release notes and using tags would be a nice improvement, unfortunately I don't have the time to roll that into this PR, but happy helping anywhere I'm needed down the line.

**Comment by EEParker** [comment]
Looks good, thanks for the help.

**Comment by EEParker** [comment]
I have one small change I want to make to this, but it can be done after the merge. I want to add a separate ci for automatic releases on tags. Dev/main/master can use the branch name and build number for the pre release version. I

**Comment by EEParker** [comment]
@VictorioBerra I have added my changes to this PR. I was able to test `Build.ps1` successfully locally, but we will have to run it here to see if the actions work.

**Comment by EEParker** [comment]
image

**Comment by nblumhardt** [comment]
Good luck with the project, @EEParker and @VictorioBerra. I'll tune out regular traffic on this repo, but please `@` me anytime I can help :-)

**Comment by EEParker** [comment]
Thank you for your assistance!

**Comment by VictorioBerra** [comment]
Thank you!!

**Pull request merged**